frontend/src/composables/use-analytics.ts

```diff
@@ -1,68 +1,7 @@
-import { computed, onMounted } from "vue"
 import { useContext } from "@nuxtjs/composition-api"
 
-import type { Events, EventName } from "~/types/analytics"
-import { useUiStore } from "~/stores/ui"
-import { useFeatureFlagStore } from "~/stores/feature-flag"
-
-import { log } from "~/utils/console"
-
-/**
- * The `ctx` parameter must be supplied if using this composable outside the
- * bounds of the composition API.
- */
 export const useAnalytics = () => {
-  const { $plausible } = useContext()
-  const uiStore = useUiStore()
-  const featureFlagStore = useFeatureFlagStore()
-
-  onMounted(() => {
-    featureFlagStore.syncAnalyticsWithLocalStorage()
-  })
-
-  /**
-   * the Plausible props that work identically on the server-side and the
-   * client-side; This excludes props that need `window`.
-   */
-  const isomorphicProps = computed(() => ({
-    breakpoint: uiStore.breakpoint,
-  }))
-
-  /**
-   * the Plausible props that work only on the client-side; This only includes
-   * props that need `window`.
-   */
-  const windowProps = computed(() =>
-    window
-      ? {
-          width: window.innerWidth,
-          height: window.innerHeight,
-        }
-      : {}
-  )
-
-  /**
-   * Send a custom event to Plausible. Mandatory props are automatically merged
-   * with the event-specific props.
-   *
-   * @param name - the name of the event being recorded
-   * @param payload - the additional information to record about the event
-   */
-  const sendCustomEvent = <T extends EventName>(
-    name: T,
-    payload: Events[T]
-  ) => {
-    log(`Analytics event: ${name}`, payload)
-    $plausible.trackEvent(name, {
-      props: {
-        ...isomorphicProps.value,
-        ...windowProps.value,
-        ...payload,
-      },
-    })
-  }
+  const { $sendCustomEvent } = useContext()
```

Comment:
I don't think this will work when called by the store or the media service, outside of the `setup` function.  Currently, the places where I think this could break is the other composables: `use-sensitive-media` and `use-search`.
It would probably also break the search response time and error handling events in the other open PRs.

Reply:
Why would it break in other composables? They also have access to nuxt context, no? Do you have a reproduction of it breaking somewhere?

Reply:
I think I tried using the "Centralise frontend errors" (which calls analytics from store's media fetching, if I remember correctly) and the "Search response times" (which cals analytics from a data service), and they failed there. We currently don't call analytics from the server or from data services, that's why tests should not be affected at all.

Reply:
@sarayourfriend, I tried re-adding the "Search response time" event (https://github.com/WordPress/openverse/tree/fix/analytics-plugin-add-search-response-time), and it does work with this code. I did have to mock the `useContext` calls in the unit tests, so maybe we should also add an e2e test to make sure it works. I'm going to approve this PR, glad that it took so few changes!